examples/argo.py

```diff
@@ -0,0 +1,252 @@
+"""ARGO: An Auto-Tuning Runtime System for Scalable GNN Training on Multi-Core Processor
+--------------------------------------------
+Graph Neural Network (GNN) training suffers from low scalability on multi-core CPUs.
+Specificially, the performance often caps at 16 cores, and no improvement is observed when applying more than 16 cores.
+ARGO is a runtime system that offers scalable performance by overlapping the computation and communication during GNN training.
+With ARGO enabled, we are able to scale over 64 cores, allowing ARGO to speedup GNN training (in terms of epoch time) by up to 5.06x and 4.54x on a Xeon 8380H and a Xeon 6430L, respectively.
+--------------------------------------------
+Paper Link: https://arxiv.org/abs/2402.03671
+"""
+
+import time
+from typing import Callable, Tuple
+
+import dgl.multiprocessing as dmp
```

Comment:
Is this needed? We don't want to have a `dgl` dependency in PyG. Can't we use Python/PyTorch for this?

Reply:
replace with torch.multiprocessing works in my use case